src/test/java/net/snowflake/client/jdbc/MaxLobSizeLatestIT.java

```diff
@@ -0,0 +1,253 @@
+  private static int mediumLobSize = largeLobSize / 2;
+  private static int originLobSize = mediumLobSize / 2;
+  private static int smallLobSize = 16;
+
+  @BeforeClass
+  public static void setUp() {
+    // TODO: May need to increase MAX_JSON_STRING_LENGTH_JVM once we increase the max LOB size.
+    System.setProperty(
+        ObjectMapperFactory.MAX_JSON_STRING_LENGTH_JVM, Integer.toString(45_000_000));
+  }
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    int[] lobSizes =
+        new int[] {smallLobSize, originLobSize, mediumLobSize, largeLobSize, maxLobSize};
+    String[] resultFormats = new String[] {"Arrow", "JSON"};
+    List<Object[]> ret = new ArrayList<>();
+    for (int i = 0; i < lobSizes.length; i++) {
+      for (int j = 0; j < resultFormats.length; j++) {
+        ret.add(new Object[] {lobSizes[i], resultFormats[j]});
+      }
+    }
+    return ret;
+  }
+
+  private final int lobSize;
+
+  private final String resultFormat;
+
+  public MaxLobSizeLatestIT(int lobSize, String resultFormat) {
+    this.lobSize = lobSize;
+    this.resultFormat = resultFormat;
+  }
+
+  private static String tableName = "my_lob_test";
+  private static String enableMaxLobSize =
+      "alter session set FEATURE_INCREASED_MAX_LOB_SIZE_IN_MEMORY = 'ENABLED'";
+  private static String executeInsert = "insert into " + tableName + " (c1, c2, c3) values (";
+  private static String executePreparedStatementInsert = executeInsert + "?, ?, ?)";
+  private static String selectQuery = "select * from " + tableName;
+
+  private static String generateRandomString(int stringSize) {
+    RandomStringGenerator randomStringGenerator =
+        new RandomStringGenerator.Builder().withinRange('a', 'z').build();
+    return randomStringGenerator.generate(stringSize);
+  }
+
+  private static void setResultFormat(Connection con, String format) throws SQLException {
+    try (Statement stmt = con.createStatement()) {
+      stmt.execute("alter session set jdbc_query_result_format = '" + format + "'");
+    }
+  }
+
+  private void createTable(int lobSize, Connection con) throws SQLException {
+    try (Statement stmt = con.createStatement()) {
+      String createTableQuery =
+          "create or replace table "
+              + tableName
+              + " (c1 varchar, c2 varchar("
+              + lobSize
+              + "), c3 int)";
+      stmt.execute(createTableQuery);
+    }
+  }
+
+  private void insertQuery(String varCharValue, int intValue, Connection con) throws SQLException {
+    try (Statement stmt = con.createStatement()) {
+      stmt.executeUpdate(
+          executeInsert + "'" + varCharValue + "', '" + varCharValue + "', " + intValue + ")");
+    }
+  }
+
+  private void preparedInsertQuery(String varCharValue, int intValue, Connection con)
+      throws SQLException {
+    try (PreparedStatement pstmt = con.prepareStatement(executePreparedStatementInsert)) {
+      pstmt.setString(1, varCharValue);
```

Comment:
please as first column use some simple string e.g. "a" (in all the tests) - it's enough to have one big column